server/src/main/java/org/opensearch/snapshots/SnapshotsService.java

```diff
@@ -334,6 +336,12 @@ public ClusterState execute(ClusterState currentState) {
                         );
                     }
                 }
+
+                boolean remoteStoreIndexShallowCopy = false;
+                if (version.onOrAfter(Version.V_2_8_0)) {
```

Comment:
1. We need to use `currentState.nodes().getMinNodeVersion()` instead of the `minCompatibilityVersion`. Also we should add a warning log if the flag is set in repository settings with a node below 2.9 that the `setting is ignored and will be used when all nodes are moved to new version`